graphql/resolve/mutation_rewriter.go

```diff
@@ -600,46 +602,48 @@ func (urw *UpdateRewriter) Rewrite(
 
 	upsertQuery := RewriteUpsertQueryFromMutation(m, authRw, MutationQueryVar, m.Name(), "")
 	srcUID := MutationQueryVarUID
-
-	if setArg == nil && delArg == nil {
+	objDel, okDelArg := delArg.(map[string]interface{})
+	objSet, okSetArg := setArg.(map[string]interface{})
+	if (setArg == nil || (len(objSet) == 0 && okSetArg)) && (delArg == nil || (len(objDel) == 0 && okDelArg)) {
```

Comment:
May be add a small comment here on why this is done.

Reply:
added.